larq_compute_engine/tflite/tests/bconv2d_test.cc

```diff
@@ -496,18 +496,25 @@ void runTest(const TestParam& param) {
   const auto is_reference_registration =
       (registration == compute_engine::tflite::Register_BCONV_2D_REF);
 
-  if (padding == Padding_SAME &&
-      (is_reference_registration || activation == ActivationFunctionType_RELU ||
-       write_bitpacked_output || int8_output)) {
-    // Zero-padding is not supported in combination with:
-    // - The reference implementation
-    // - Fused ReLu
-    // - Writing bitpacked output
-    // - Int8 output
-    // We could use `EXPECT_DEATH` here but it is extremely slow. Therefore we
-    // have a separate test below, and here we just skip.
-    GTEST_SKIP();
-    return;
+  const bool reference_supports_zero_padding =
+      is_reference_registration && (input_depth % 2 == 0);
+
+  if (padding == Padding_SAME && !reference_supports_zero_padding) {
+    if (groups > 1 || activation == ActivationFunctionType_RELU ||
+        write_bitpacked_output || int8_output) {
```

Comment:
I think `groups > 1` is no longer a constraint?

How do you feel about breaking it down slightly more, to something like:
```
if ZERO_PADDING {
    if (is_reference) {
        if (!even_c_i || has_relu) SKIP()
    } else {
        if (!float_output || has_relu) SKIP()
    }
}
```

I think that might make it a bit clearer what's going on.